### DISCUSSION TITLE
Improved request-sequence testing without “links”

### DISCUSSION THREAD

**Discussion by TheBigRoomXXL** [schemathesis/schemathesis#1571] [General]
As a direction for future research, you wrote in your paper :

> _Improved request-sequence testing without “links”._ Schemathesis is often limited by the absence of OpenAPI “links” describing how data from one response can be used to make further requests. Investigating [3, 30]-style heuristics or ways to learn links from traffic records could substantially improve performance.

I am indeed very interested by that kind of feature. Is there some work on that subject already being done? Where should I start to contribute toward that goal?

**Comment by Stranger6667** [comment]
Oh, I missed your message and a lot of time has passed since. I have some private implementation on the Schemathesis.io side and at the moment I want to keep it as a service-only feature, that may just generate links on demand (accessible via CLI). Generally, I think the best approach would be to start with the current literature and decide on the architecture. There are a lot of useful tricks for detecting such links, but I don't know yet how to combine them and whether such combinations would be effective.
